libcaf_net/caf/net/http/client.cpp

```diff
@@ -206,10 +206,49 @@ class client_impl : public client {
       input.subspan(payload_len_);
       mode_ = mode::read_header;
       return consumed;
+    } else if (mode_ == mode::read_chunks) {
+      auto res = v1::parse_chunk(input);
+      if (!res) {
+        // No error code signals we didn't receive enough data.
+        if (!res.error())
+          return consumed;
+        abort_and_shutdown(res.error());
+        return -1;
+      }
+      auto [chunk_size, remainder] = *res;
+      // Protect early against payloads that exceed the maximum size.
+      if (chunk_size + buffer_.size() >= max_response_size_) {
+        abort_and_shutdown("Payload exceeds maximum size.");
+        return -1;
+      }
+      if (remainder.size() < chunk_size + 2) {
+        // Configure the policy for the next call to consume.
+        // Await exactly chunk line length + chunk_size bytes + crlf.
+        const auto awaited = input.size() - remainder.size() + chunk_size + 2;
+        down_->configure_read(receive_policy::exactly(awaited));
+        return consumed;
+      }
+      // Reset the policy from the previous call to consume.
+      down_->configure_read(receive_policy::up_to(max_response_size_));
+      consumed += input.size() - remainder.size() + chunk_size + 2;
+      // End of chunk encoded request comes with a zero length chunk.
+      if (chunk_size == 0) {
+        if (!invoke_upper_layer(buffer_))
+          return -1;
+        mode_ = mode::read_header;
+        buffer_.clear();
+        return consumed;
+      }
+      buffer_.reserve(buffer_.size() + chunk_size);
```

Comment:
```suggestion
```

`reserve` is primarily useful for avoiding re-allocations with smaller inserts or things like `emplace_back`. We insert the full range below in one go, so there's no need for a separate `reserve` here.